core/group/service.go

```diff
@@ -78,7 +79,7 @@ func (s Service) List(ctx context.Context, flt Filter) ([]Group, error) {
 }
 
 func (s Service) Update(ctx context.Context, grp Group) (Group, error) {
-	if grp.ID != "" {
+	if !str.IsStringEmpty(grp.ID) {
```

Comment:
just a side note on this `str.IsStringEmpty` function. It is advertised and exposed as a public "generic" function. It has no document comment as well. But it does a `TrimSpace()` before doing an empty check. So it's not really generic. And using `!str.IsStringEmtpy()` in place of `grp.ID != ""` is actually changing the behaviour slightly (In this particular case, it is for the better). 

Usually it's better to avoid trying to extract everything into "reusable" functions. Just doing `strings.TrimSpace(val) != ""` is good enough as well.

Reply:
I think you have a fair point on that. I think this one is not necessary to be extracted out. Keeping that inline would make this more readable. Thanks for the input @spy16